src/browser/BrowserService.cpp

```diff
@@ -590,13 +597,42 @@ BrowserService::searchEntries(const QSharedPointer<Database>& db, const QString&
         if ((entryQUrl.port() > 0 && entryQUrl.port() != qUrl.port())
             || (browserSettings()->matchUrlScheme() && !entryScheme.isEmpty()
                 && entryScheme.compare(qUrl.scheme()) != 0)) {
-            continue;
+            return false;
         }
 
         // Filter to match hostname in URL field
         if ((!entryUrl.isEmpty() && hostname.contains(entryUrl))
             || (matchUrlScheme(entryUrl) && hostname.endsWith(entryQUrl.host()))) {
-            entries.append(entry);
+            return true;
+        }
+        return false;
+    };
+
+    // Search host from URL fields
+    for (Entry* entry : EntrySearcher().search(baseDomain(hostname), rootGroup)) {
+        if (!handleURL(entry->url())) {
+            continue;
+        }
+
+        entries.append(entry);
+    }
```

Comment:
const auto&

Reply:
It's QList<Entry*> everywhere. We really want to modify everything to QList<const Entry*>?

Reply:
Standby I think my comment went to the wrong line